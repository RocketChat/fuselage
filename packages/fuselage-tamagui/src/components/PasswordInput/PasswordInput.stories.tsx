import type { Meta, StoryObj } from '@storybook/react';
import PasswordInput from './PasswordInput';

const meta = {
  title: 'Inputs/PasswordInput',
  component: PasswordInput,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    secondary: {
      control: 'boolean',
      description: 'Whether to use secondary styling',
    },
    small: {
      control: 'boolean',
      description: 'Whether to use small size',
    },
    disabled: {
      control: 'boolean',
      description: 'Whether the input is disabled',
    },
    error: {
      control: 'text',
      description: 'Error message to display',
    },
    placeholder: {
      control: 'text',
      description: 'Placeholder text',
    },
    value: {
      control: 'text',
      description: 'Input value',
    },
  },
} satisfies Meta<typeof PasswordInput>;

export default meta;
type Story = StoryObj<typeof PasswordInput>;

export const Default: Story = {
  args: {
    'aria-label': 'password',
  },
};

export const WithValue: Story = {
  args: {
    'aria-label': 'password',
    value: 'mypassword123',
  },
};

export const WithPlaceholder: Story = {
  args: {
    'aria-label': 'password',
    placeholder: 'Enter your password',
  },
};

export const Invalid: Story = {
  args: {
    'aria-label': 'password',
    error: 'Password is required',
  },
};

export const Disabled: Story = {
  args: {
    'aria-label': 'password',
    disabled: true,
    value: 'mypassword123',
  },
};

export const Small: Story = {
  args: {
    'aria-label': 'password',
    small: true,
    placeholder: 'Enter password',
  },
};

export const WithValueSmall: Story = {
  args: {
    'aria-label': 'password',
    small: true,
    value: 'mypassword123',
  },
};
