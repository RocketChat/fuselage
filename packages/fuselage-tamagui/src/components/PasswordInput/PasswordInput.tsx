import { useToggle } from '@rocket.chat/fuselage-hooks';
import type { ComponentProps, Ref } from 'react';
import { forwardRef } from 'react';

import { Icon } from '../Icon';
import { InputBox } from '../InputBox';

// TODO: fix a11y issues

type PasswordInputProps = Omit<ComponentProps<typeof InputBox>, 'type'>;

const PasswordInput = forwardRef<HTMLInputElement, PasswordInputProps>(
  function PasswordInput(props, ref) {
    const [hidden, toggle] = useToggle(true);
    const handleAddonClick = () => {
      toggle();
    };

    return (
      <InputBox
        type={hidden ? 'password' : 'text'}
        addon={
          <Icon
            name={hidden ? 'eye-off' : 'eye'}
            size={20}
            onClick={handleAddonClick}
          />
        }
        ref={ref}
        {...props}
      />
    );
  }
);

PasswordInput.displayName = 'PasswordInput';

export { PasswordInput };
export default PasswordInput;
