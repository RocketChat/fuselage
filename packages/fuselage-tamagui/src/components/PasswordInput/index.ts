export { default, PasswordInput } from './PasswordInput';
