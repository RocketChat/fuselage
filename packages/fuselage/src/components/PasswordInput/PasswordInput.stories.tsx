import React from 'react';

import { Icon, PasswordInput } from '../..';
import PropsVariation from '../../.storybook/PropsVariation';

export default {
  title: 'Forms/Inputs/PasswordInput',
  component: PasswordInput,
  parameters: { jest: ['PasswordInput.spec.tsx'] },
};

export const Default = () => <PasswordInput />;

export const WithValue = () => <PasswordInput defaultValue='password' />;

export const WithIconAddon = () => (
  <PasswordInput addon={<Icon name='send' size={20} />} />
);

export const Invalid = () => <PasswordInput error='Error' />;

export const Disabled = () => <PasswordInput disabled />;

export const WithPlaceholder = () => (
  <PasswordInput placeholder='Placeholder' />
);

export const States = () => (
  <PropsVariation
    component={PasswordInput}
    common={{ onChange: () => undefined }}
    xAxis={{
      'default': {},
      'with placeholder': { placeholder: 'Placeholder' },
      'with value': { value: 'Value' },
      'with icon': { addon: <Icon name='key' size='x20' />, value: 'Value' },
    }}
    yAxis={{
      'default': {},
      'hover': { className: 'hover' },
      'active': { className: 'active' },
      'focus': { className: 'focus' },
      'disabled': { disabled: true },
      'errored': { error: 'Error' },
      'errored + hover': { className: 'hover', error: 'Error' },
      'errored + active': { className: 'active', error: 'Error' },
      'errored + focus': { className: 'focus', error: 'Error' },
    }}
  />
);
