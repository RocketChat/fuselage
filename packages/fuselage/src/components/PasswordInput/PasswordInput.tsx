import { useToggle } from '@rocket.chat/fuselage-hooks';
import React, { ComponentProps, forwardRef, Ref } from 'react';

import { Icon } from '../Icon';
import { InputBox } from '../InputBox';

// TODO: fix a11y issues

type PasswordInputProps = Omit<ComponentProps<typeof InputBox>, 'type'>;

const PasswordInput = forwardRef(function PasswordInput(
  props: PasswordInputProps,
  ref: Ref<HTMLInputElement>
) {
  const [hidden, toggle] = useToggle(true);
  const handleAddonClick = () => {
    toggle();
  };

  return (
    <InputBox
      type={hidden ? 'password' : 'text'}
      addon={
        <Icon
          name={hidden ? 'eye-off' : 'eye'}
          size={20}
          onClick={handleAddonClick}
        />
      }
      ref={ref}
      {...props}
    />
  );
});

export default PasswordInput;
