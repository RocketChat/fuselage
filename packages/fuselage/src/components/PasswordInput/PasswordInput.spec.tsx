import React from 'react';
import ReactDOM from 'react-dom';

import PasswordInput from '.';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<PasswordInput />, div);
  ReactDOM.unmountComponentAtNode(div);
});
