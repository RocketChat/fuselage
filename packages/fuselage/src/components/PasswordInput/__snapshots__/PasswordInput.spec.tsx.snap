// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PasswordInput Component] renders without crashing 1`] = `
<body>
  <div>
    <label
      class="rcx-box rcx-box--full rcx-label rcx-box rcx-box--full rcx-box--animated rcx-input-box__wrapper"
    >
      <input
        aria-label="password"
        class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--undecorated rcx-input-box--type-password rcx-input-box"
        size="1"
        type="password"
      />
      <span
        class="rcx-box rcx-box--full rcx-input-box__addon"
      >
        <i
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-icon--name-eye-off rcx-icon rcx-css-1bepdyv"
        >
          î€«
        </i>
      </span>
    </label>
  </div>
</body>
`;
